### PR TITLE
misc: add design architecture + simplify code part

### PR DESCRIPTION
**Pull request by Hugo-Rybinski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,11 +40,48 @@ Build a small application exposing the following endpoints through a REST API:
...
+ * The complexity & scalability of your algorithm
+ * The readability of your code (including readability of your tests)
+ * The correctness of the outputs of the API
+ * You're ability to share your design choices and clearly weight pros & cons of alternative solutions
+ * The overall quality of your written technical communication
+
+
+### Part 2: System design
+
+Describe briefly how you would build at scale and in the cloud (preferably with GCP) an API to power a simplified page of our Search Analytics:
```
Comment (mvonwyl) [inline README.md:52]: If we want to judge them mainly on that part, I'd recommend removing "briefly"
```suggestion
Describe how you would build at scale and in the cloud (preferably with GCP) an API to power a simplified page of our Search Analytics:
```

File: README.md [hunk 2]
```diff
@@ -66,11 +40,48 @@ Build a small application exposing the following endpoints through a REST API:
...
+* Each click is tied to a specific search in the log with a `queryID`
+* Click events are accepted by the API if they are reiceved up to 15 minutes after the search
+* Customer may request up to 3 month of timerange in the Analytics
+
+Describe the eventual databases that you would setup and their content, the eventual packaged services you would use, and the different processes you would create. For each logical brick, explain their goal and how do they communicate together.
+
+#### Evaluation Criteria
+
+* Please push it along with your code on GitHub repository or in the archive
+* Stay concise (maximum 1 page)
```
Comment (mvonwyl) [inline README.md]: Again here, I'd be happy with 2-3 pages and not being too concise, especially if it includes several diagrams.

File: README.md (hunk 1 above)
Comment (Hugo-Rybinski) [reply README.md:52]: We want them to be concise and not handling us tens of pages of specification (that's why we added a size limit). The goal here is to use their design description as a base for our discussion during the assignment review where we can challenge it/add further constraints to see how they adapt (like we do during a regular architecture design specification)

File: README.md [hunk 3]
```diff
@@ -66,11 +40,48 @@ Build a small application exposing the following endpoints through a REST API:
 
 We'll evaluate:
  
- * the complexity & scalability of your algorithm
- * the readability of your code (including readability of your tests)
- * the correctness of the outputs of the API
- * you're ability to share your design choices and clearly weight pros & cons of alternative solutions
- * the overall quality of your written technical communication 
+ * The complexity & scalability of your algorithm
+ * The readability of your code (including readability of your tests)
```
Comment (slashvar) [inline README.md:44]: If I'm correct this is the first time we mention tests in the assignments, we should at least indicate that we expect test in the enumeration above, no ?

Comment (Hugo-Rybinski) [reply README.md:44]: It's part of our version of the assignment since a while 🤔

Comment (slashvar) [reply README.md:44]: So, I've read it from bottom to top and when reaching that sentence it stroke me that we didn't asked for it.

I'm probably biased by my teaching experience where anything you don't ask for explicitly is never done (any ambiguity is interpreted toward the least effort ... ) but what apply to students may not apply to job candidates.

Comment (mvonwyl) [reply README.md:44]: From my experience on assignment, candidates tend to do the same (if it's not asked, it doesn't exist) a lot more than we'd expect (especially for things like commenting, tests, ...). It really depends if you want to make it an "implicit filter" by only taking people senior enough to put them even when not asked.

Comment (rayrutjes) [reply README.md:44]: 👍 let's make it explicit, saying something along the lines of "Add the tests you deem necessary to assert the correct behavior of your solution."

File: README.md (hunk 1 above)
Comment (rayrutjes) [reply README.md:52]: Let's go with something like "at most a page of text and as many diagrams as you deem necessary for the good comprehension of your design."

**Pull request merged**